dataactbroker/handlers/fileHandler.py

```diff
@@ -552,6 +552,9 @@ def load_d_file(self, url, upload_name, timestamped_name, job_id, is_local, resp
 
             logger.debug('Marking job id of %s', job_id)
             mark_job_status(job_id, "finished")
+            # only need to close it if we threaded it in the first place
+            if not self.isLocal:
```

Comment:
The finally block will run regardless of the return statement or the exception in the preceding blocks. Here's the documentation that I'm referencing: https://docs.python.org/3/reference/compound_stmts.html#the-try-statement